### PR TITLE
In-game Armor sprite change

### PR DESCRIPTION
**Pull request by lispcoc**
#### Summary
Features "In-game Armor sprite change"

#### Purpose of change
Allows you to look the way you want no matter what armor you wear.

#### Describe the solution
"Change sprite" option in player status menu. you can select sprite from item you have.

#### Testing
Game starts. Sprite is able to be overriden.  Restore default sprite successfully.

#### Additional context
image

### CODE REVIEW

File: src/armor_layers.cpp [hunk 1]
```diff
@@ -1031,6 +1034,36 @@ void outfit::sort_armor( Character &guy )
...
+                if( loc && loc.get_item() ) {
+                    const item *i = loc.get_item();
+                    const std::string variant = i->has_itype_variant() ? i->itype_variant().id : "";
+                    tmp_worn[leftListIndex]->set_var( "sprite_override", i->typeId().str() );
+                    tmp_worn[leftListIndex]->set_var( "sprite_override_variant", variant );
+                }
+            } else if( menu.ret == 1 ) {
+                tmp_worn[leftListIndex]->erase_var( "sprite_override" );
+                tmp_worn[leftListIndex]->erase_var( "sprite_override_variant" );
+            }
```
Comment (mqrause) [inline src/armor_layers.cpp:1066]: Might be just a drop in the ocean, but it'd be great if you would extract that in its own method, `outfit::sort_armor` is already far too big as is.

Comment (lispcoc) [reply src/armor_layers.cpp:1066]: Should I make new function like outfit::change_sprite, or make new interface outside sorting armor menu?

File: data/raw/keybindings.json [hunk 2]
```diff
@@ -588,6 +588,13 @@
     "name": "Toggle armor visibility on character sprite",
     "bindings": [ { "input_method": "keyboard_any", "key": "H" } ]
   },
+  {
+    "type": "keybinding",
+    "id": "CHANGE_SPRITE",
+    "category": "SORT_ARMOR",
+    "name": "Change sprite of armor.",
+    "bindings": [ { "input_method": "keyboard_any", "key": "C" } ]
```
Comment (RenechCDDA) [inline data/raw/keybindings.json:596]: I don't think this should be available as a "top-level" keybinding. Check out #59677 for the reasoning.

I think it should be a key found within the status menu (@ screen)

File: src/armor_layers.cpp (hunk 1 above)
Comment (mqrause) [reply src/armor_layers.cpp:1066]: It doesn't have to be a member of the outfit class, a static method in the same file is enough. Unless you have some plans to make this functionality available from other places, too.

File: data/raw/keybindings.json (hunk 2 above)
Comment (Qrox) [reply data/raw/keybindings.json]: In addition, this (and `SORT_ARMOR` above) should be changed to specify the modifier key for the key code mode like this:

```suggestion
    "bindings": [ { "input_method": "keyboard_char", "key": "C" }, { "input_method": "keyboard_code", "key": "c", "mod": [ 
"shift" ] } ]
```

File: data/raw/keybindings.json [hunk 3]
```diff
@@ -4251,6 +4251,13 @@
     "id": "morale",
     "bindings": [ { "input_method": "keyboard_any", "key": "m" } ]
   },
+  {
+    "type": "keybinding",
+    "id": "CHANGE_ARMOR_SPRITE",
+    "category": "PLAYER_INFO",
+    "name": "Change sprite of armor.",
+    "bindings": [ { "input_method": "keyboard_any", "key": "W" } ]
```
Comment (Qrox) [inline data/raw/keybindings.json:4259]:
```suggestion
    "bindings": [ { "input_method": "keyboard_char", "key": "W" }, { "input_method": "keyboard_code", "key": "w", "mod": [ "shift" ] } ]
```

Comment (lispcoc) [reply data/raw/keybindings.json]: Thank you! fixed

File: data/raw/keybindings.json (hunk 2 above)
Comment (lispcoc) [reply data/raw/keybindings.json]: moved to status menu.

File: data/raw/keybindings.json [hunk 4]
```diff
@@ -4251,6 +4251,13 @@
     "id": "morale",
     "bindings": [ { "input_method": "keyboard_any", "key": "m" } ]
   },
+  {
+    "type": "keybinding",
+    "id": "CHANGE_ARMOR_SPRITE",
+    "category": "PLAYER_INFO",
+    "name": "Change sprite of armor.",
+    "bindings": [ { "input_method": "keyboard_any", "key": "W" }, { "input_method": "keyboard_code", "key": "w", "mod": [ "shift" ] } ]
```
Comment (Qrox) [inline data/raw/keybindings.json:4259]:
```suggestion
    "bindings": [ { "input_method": "keyboard_char", "key": "W" }, { "input_method": "keyboard_code", "key": "w", "mod": [ "shift" ] } ]
```

**Comment by alp7292** [comment]
i really like this change. dont care about realism i dont want to look like hobo because clothes i wear is best for warmth and protection.

**Comment by alp7292** [comment]
can you change the sprite only for the items you wear or also the items you have in your inventory. cuz if its only for the ones you wear you cant change shoes

**Comment by lispcoc** [comment]
> can you change the sprite only for the items you wear or also the items you have in your inventory. cuz if its only for the ones you wear you cant change shoes and helmets

I think shoes and helmets already changeable, but its more useful. I will change like that.

**Comment by bombasticSlacks** [comment]
You have a CLANG tidy error to resolve

**Comment by lispcoc** [comment]
> You have a CLANG tidy error to resolve

It seems to be fixed, Thanks!

**Comment by bombasticSlacks** [comment]
Thanks, and thanks for the contributions 😄

**Pull request merged**